src/index.ts

```diff
@@ -159,11 +166,40 @@ export class MondayRedisService extends BaseConnector {
               this.serialize(item[title], title),
             )
           ),
-          this.redis.hset(this.namesKey, item.name, item.id)
+          this.redis.hset(this.namesKey, item.name, item.id),
         ]))
-        .flat()
+          .flat()
       )
     }
+
+    this.boardId = await this.getBoardId()
+    this.monday.on(
+      {
+        type: 'ChangeColumnValue',
+        boardId: this.boardId,
+      },
+      async (event) => {
+        const { boardId, itemId, columnTitle, value: { value } } = event
+        if (this.boardId === parseInt(boardId, 10)) {
```

Comment:
Since we already have the value here from the callback, I think your original approach would be better. Simply update Redis instead of dealing with a special "INVALIDATED" value. This will remove the need for the complex changes to "incrColumVlaue" which raise a myriad of questions around concurrency and atomicity.

Also, at this point we need to remove any changes related to this item/column from the changes in `this.changesKey`. I suggest to add to "destageChances" a parameter with an item/column pair to ignore and call it here. This will make sure that the changes list is handled atomically, as per the first line of distageChanges

Reply:
totally agree with the first block, I wanted to suggest the same.
About changesKey, I think it's safer to just update the list. As an increment can happened between 'new value coming from Monday via event' and 'next interval sync with Monday', so it's just simpler.